### PR TITLE
TINKERPOP-2392 Improved GLV starting docs

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-2392

This issue was about javascript but really might have applied to all GLVs. With the recent change of ae95dbe9c33b380555ee32716e56ec12b56c3f6c it seemed to make sense to expand this change to all GLVs and to improve upon it. I could have probably just CTR'd this in but thought it would be better to get some review on sample variety and syntax (as these changes will affect artifact repositories descriptions...nuget, pypi, npm...and I'm not sure there is a nice way to validate links/changes without a release deployment.

VOTE +1

### CODE REVIEW

**Comment by spmallette** [comment]
@aboudreault if you have a moment to take a look at the python related doc changes to see if they would make sense to new python developer, that would be appreciated.

**Comment by jsumners** [comment]
I think these additions go a long way toward making the documentation clearer for secondary languages like JavaScript.

**Comment by gugacavalieri** [comment]
Looks like a big improvement :orange_heart:

File: gremlin-python/src/main/jython/README.rst [hunk 1]
```diff
@@ -54,8 +54,65 @@ remote graph:
...
-Please see the `reference documentation <http://tinkerpop.apache.org/docs/current/reference/#gremlin-python>`_
-at Apache TinkerPop for more information on usage.
+-----------------
+Sample Traversals
+-----------------
+
+The Gremlin language allows users to write highly expressive graph traversals and has a broad list of functions that
+cover a wide body of features. The `Reference Documentation <https://tinkerpop.apache.org/docs/current/reference/#graph-traversal-steps>`_
+describes these functions and other aspects of the TinkerPop ecosystem including some specifics on
+`Gremlin in Python <http://tinkerpop.apache.org/docs/current/reference/#gremlin-javascript>`_ itself. Most of the
```
Comment (aboudreault) [inline gremlin-python/src/main/jython/README.rst:64]: to fix: use #gremlin-python here?

**Comment by aboudreault** [comment]
Yep, good docs changes. Thanks Stephen!

File: gremlin-python/src/main/jython/README.rst (hunk 1 above)
Comment (spmallette) [reply gremlin-python/src/main/jython/README.rst]: yes - thank you. i missed that link.

File: gremlin-dotnet/README.md [hunk 2]
```diff
@@ -33,8 +33,22 @@ operating systems and with different .NET frameworks, such as .NET Framework and
...
+Gremlin variant for that language.
+
+**NOTE** that versions suffixed with "-rc" are considered release candidates (i.e. pre-alpha, alpha, beta, etc.) and thus 
+for early testing purposes only.
 
 [tk]: http://tinkerpop.apache.org
 [gremlin]: http://tinkerpop.apache.org/gremlin.html
-[docs]: http://tinkerpop.apache.org/docs/current/reference/#gremlin-DotNet
\ No newline at end of file
+[docs]: http://tinkerpop.apache.org/docs/current/reference/#gremlin-dotnet
```
Comment (FlorianHockmann) [inline gremlin-dotnet/README.md:51]: Just a nit and not really related to this PR, but we can also upgrade these links to HTTPS now.

**Comment by spmallette** [comment]
I addressed all of the @FlorianHockmann suggestions - good catch with using "https" for links. I'd done that consistently elsewhere in our docs long ago but must have missed these somehow.

**Review by FlorianHockmann — APPROVED** [review]
VOTE +1

**Review by jorgebay — COMMENTED** [review]
lgtm, just a small remark below.

**Review by jorgebay — APPROVED** [review]
VOTE +1

**Pull request merged**